### PR TITLE
HTTP2: Add GrpcNetworkStreamState metadata

### PR DESCRIPTION
**Pull request by yashykt**
HTTP2: Add GrpcNetworkStreamState metadata for calls that are not sent on wire and for those that are not seen by server

### CODE REVIEW

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc
Comment (markdroth) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.cc:1108]: It looks like we're no longer setting the status to UNAVAILABLE here, but I see that the tests are still checking for this, and they are presumably passing, so it must be happening somewhere else.  Where is this happening now?

**Review by markdroth — APPROVED** [review]
This looks great to me, but please wait for Craig to review before merging.

Thanks for doing this!

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc
Comment (yashykt) [reply src/core/ext/transport/chttp2/transport/chttp2_transport.cc:1108]: https://github.com/grpc/grpc/blob/cf81e41162603d229d0c8ef05f137a0afbe1f3c6/src/core/ext/transport/chttp2/transport/chttp2_transport.cc#L1054

goaway_error sets the status already

Comment (markdroth) [reply src/core/ext/transport/chttp2/transport/chttp2_transport.cc:1108]: Got it, thanks!

**Review by ctiller — APPROVED** [review]

**Pull request merged**
